pages_deployment.go

```diff
@@ -0,0 +1,202 @@
+
+	uri := fmt.Sprintf("/accounts/%s/pages/projects/%s/deployments", accountID, projectName)
+	if len(v) > 0 {
+		uri = fmt.Sprintf("%s?%s", uri, v.Encode())
+	}
+
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return []PagesProjectDeployment{}, ResultInfo{}, err
+	}
+	var r pagesDeploymentListResponse
+	err = json.Unmarshal(res, &r)
+	if err != nil {
+		return []PagesProjectDeployment{}, ResultInfo{}, errors.Wrap(err, errUnmarshalError)
+	}
+	return r.Result, r.ResultInfo, nil
+}
+
+// GetPagesDeploymentInfo returns a deployment for a Pages project.
+//
+// API reference: https://api.cloudflare.com/#pages-deployment-get-deployment-info
+func (api *API) GetPagesDeploymentInfo(ctx context.Context, accountID string, projectName string, deploymentID string) (PagesProjectDeployment, error) {
+	uri := fmt.Sprintf("/accounts/%s/pages/projects/%s/deployments/%s", accountID, projectName, deploymentID)
+
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return PagesProjectDeployment{}, err
+	}
+	var r pagesDeploymentResponse
+	err = json.Unmarshal(res, &r)
+	if err != nil {
+		return PagesProjectDeployment{}, errors.Wrap(err, errUnmarshalError)
+	}
+	return r.Result, nil
+}
+
+// GetPagesDeploymentStageLogs returns the logs for a Pages deployment stage.
+//
+// API reference: https://api.cloudflare.com/#pages-deployment-get-deployment-stage-logs
+func (api *API) GetPagesDeploymentStageLogs(ctx context.Context, accountID string, projectName string, deploymentID string, stageName string, sizeOpts SizeOptions) (PagesDeploymentStageLogs, error) {
+	v := url.Values{}
+	if sizeOpts.Size > 0 {
+		v.Set("size", strconv.Itoa(sizeOpts.Size))
+	}
+	if sizeOpts.Before > 0 {
+		v.Set("before", strconv.Itoa(sizeOpts.Before))
+	} else if sizeOpts.After > 0 {
+		v.Set("after", strconv.Itoa(sizeOpts.After))
+	}
+
+	uri := fmt.Sprintf("/accounts/%s/pages/projects/%s/deployments/%s/history/%s/logs", accountID, projectName, deploymentID, stageName)
+	if len(v) > 0 {
+		uri = fmt.Sprintf("%s?%s", uri, v.Encode())
+	}
+
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return PagesDeploymentStageLogs{}, err
+	}
+	var r pagesDeploymentStageLogsResponse
+	err = json.Unmarshal(res, &r)
+	if err != nil {
+		return PagesDeploymentStageLogs{}, errors.Wrap(err, errUnmarshalError)
+	}
+	return r.Result, nil
+}
+
+// DeletePagesDeployment deletes a Pages deployment.
+//
+// API reference: https://api.cloudflare.com/#pages-deployment-delete-deployment
+func (api *API) DeletePagesDeployment(ctx context.Context, accountID string, projectName string, deploymentID string) error {
+	uri := fmt.Sprintf("/accounts/%s/pages/projects/%s/deployments/%s", accountID, projectName, deploymentID)
+
+	_, err := api.makeRequestContext(ctx, http.MethodDelete, uri, nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// CreatePagesDeployment creates a Pages production deployment.
```

Comment:
```suggestion
// CreatePagesDeployment creates a Pages deployment.
```

Reply:
The reason I'm explicit here is because you cannot create preview deployments with this endpoint _yet_. It is only production

Reply:
The reason I'm explicit here is because you cannot create preview deployments with this endpoint _yet_. It is only production